mock-data/MockRiddles.ts

```diff
@@ -1,24 +1,67 @@
-import type { CountryCode } from '../src/utils/CountryCodes';
-import type { Difficulty } from '../src/utils/Difficulty';
 import { getDifficultyObject } from '../src/utils/Difficulty';
 import { RiddleStatus } from '../src/utils/Enums';
+import type {
+	QuestionDisplayDetail,
+	QuestionUpsertDetail,
+	RiddleDisplayDetail,
+	RiddlePreview,
+	RiddleUpsertDetail
+} from '../src/utils/Types.ts';
 
-//TODO: Move type declaration to types
+const MockUpsertQuestions: QuestionUpsertDetail[] = [
+	{
+		id: 1,
+		number: 2,
+		text: 'It is blue and annoying, what is it?',
+		image: 'public/vite.svg',
+		hints: [],
+		correctAnswers: [{ text: 'Smurf' }]
+	},
+	{
+		id: 2,
+		number: 1,
+		text:
+			'    Ten centuries shall the fortress stand\n' +
+			'    Walls of spirit wrapped in walls of fire\n' +
+			'    And horned lords shall bowl their head\n' +
+			'    To one not yet born, of the darkest sire\n' +
+			'\n' +
+			'    One century of blood and strife\n' +
+			'    The moon shall darken, and none know why\n' +
+			'    The resting place at last is found\n' +
+			'    Of the Seventh, who soared so high\n' +
+			'\n' +
+			'    Last daughter of a forlorn line\n' +
+			'    Shall guide him into history\n' +
+			'    Beneath the crypts prophecies clash\n' +
+			'    The war of ancient enemies',
+		hints: [
+			{ text: 'Heroes', order: 1 },
+			{ text: 'Might & Magic', order: 2 },
+			{ text: '5. installment', order: 3 }
+		],
+		correctAnswers: [{ text: 'Smurf' }]
+	}
+];
 
-type Riddle = {
-	id: number;
-	name: string;
-	image: string;
-	state: RiddleStatus;
-	countryCode: CountryCode;
-	difficulty: Difficulty;
-};
+const MockDisplayQuestions: QuestionDisplayDetail[] = MockUpsertQuestions.map(
+	question => {
+		const { hints, ...rest } = question;
+		const newHints = hints?.map(hint => ({ ...hint, taken: false })) || [];
```

Comment:
Toto podle mě není správně. `hints` je mandatory a bude vždy pole a i prázdné pole je truthy, tedy mělo by stačit takto:
```suggestion
const newHints = hints.map(hint => ({ ...hint, taken: false }));
```

Pokud je `hints === []`, pak je i `newHints === []`.
Doufám, že se nepletu.

Plus taková drobnost, možná bych změnil název proměnné na `updatedHints`, přijde mi to více odpovídající. :)

Reply:
Souhlasím, poučení pro mě: umělá inteligence není na tak dobré úrovni, aby to odhalila :D